library/src/interactor_impl.cxx

```diff
@@ -523,6 +433,7 @@ class interactor_impl::internals
   std::map<unsigned long, std::pair<int, std::function<void()>>> TimerCallBacks;
 
   std::map<std::string, std::function<bool(const std::vector<std::string>&)>> CommandCallbacks;
+  std::map<std::pair<std::string, ModifierKeys>, std::vector<std::string>> InteractionCommands;
```

Comment:
I think the string key and the modifier are so tightly related we should have a structure like:
```cpp
struct KeyBind
{
  std::string KeyCode;
  ModifierKeys KeyMod;
}
```

That way, the code becomes much more clear than with a std::pair.  
Instead of `std::make_pair(std::move(interaction), modifiers)`, you can just do `{ std::move(interaction), modifiers }`  
And more importantly, you can do `mapKey.KeyCode`/`mapKey.KeyMod` instead of `mapKey.first`/`mapKey.second`

Reply:
i agree, the more I use pair, the less I like it.

Reply:
although its not KeyCode (or KeySym), its Interaction